### PR TITLE
[fix](delete) fix query result error after delete

### LINKED ISSUE #11425: [Bug] [daily] delete where char > xxx, result error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master
trunk-27a1911

### What's Wrong?

mysql> CREATE TABLE `ds` (
    ->   `tinyint_key` tinyint(4) NOT NULL,
    ->   `char_50_key` char(50) NOT NULL,
    ->   `character_key` varchar(500) NOT NULL
    -> ) ENGINE=OLAP
    -> AGGREGATE KEY(`tinyint_key`, `char_50_key`, `character_key`)
    -> COMMENT 'OLAP'
    -> DISTRIBUTED BY HASH(`tinyint_key`) BUCKETS 10
    -> PROPERTIES (
    -> "replication_allocation" = "tag.location.default: 3",
    -> "in_memory" = "false",
    -> "storage_format" = "V2"
    -> );
Query OK, 0 rows affected (0.03 sec)

mysql> insert into ds values(1, 'dddd', 'adgs'), (1, 'SSS', 'sk6S0'), (1, 'ttt', 'zdges');
Query OK, 3 rows affected (0.05 sec)
{'label':'insert_83f8df857b5f4c59_832de8b483dac445', 'status':'VISIBLE', 'txnId':'17869'}

mysql> select * from ds;
+-------------+-------------+---------------+
| tinyint_key | char_50_key | character_key |
+-------------+-------------+---------------+
|           1 | SSS         | sk6S0         |
|           1 | dddd        | adgs          |
|           1 | ttt         | zdges         |
+-------------+-------------+---------------+
3 rows in set (0.04 sec)

mysql> select * FROM ds where character_key < "sk6S0";
+-------------+-------------+---------------+
| tinyint_key | char_50_key | character_key |
+-------------+-------------+---------------+
|           1 | dddd        | adgs          |
+-------------+-------------+---------------+
1 row in set (0.04 sec)

mysql> delete from ds where character_key < "sk6S0";
Query OK, 0 rows affected (0.05 sec)
{'label':'delete_539bdcec-b49a-4b3e-9919-a31bce17d6a6', 'status':'VISIBLE', 'txnId':'17870'}

mysql> select * from ds;
+-------------+-------------+---------------+
| tinyint_key | char_50_key | character_key |
+-------------+-------------+---------------+
|           1 | SSS         | sk6S0         |
|           1 | dddd        | adgs          |
|           1 | ttt         | zdges         |
+-------------+-------------+---------------+
3 rows in set (0.04 sec)

### What You Expected?

delete result right

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
convert dictionary code for delete predicates.

# Proposed changes

Issue Number: close #11425 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — CHANGES_REQUESTED** [review]
Please add regression test from the issue.

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -1199,6 +1199,34 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
...
+        _convert_dict_code_for_predicate_if_necessary_impl(predicate);
+    }
+
+    for (auto predicate : _pre_eval_block_predicate) {
+        _convert_dict_code_for_predicate_if_necessary_impl(predicate);
+    }
+
+    std::set<const ColumnPredicate*> predicate_set {};
+    _opts.delete_condition_predicates->get_all_column_predicate(predicate_set);
+    for (const auto column_predicate : predicate_set) {
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:1213]: Could you please also check if  dev-1.1.2 also have this problem?

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -1199,6 +1199,34 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
...
+    for (auto predicate : _short_cir_eval_predicate) {
+        _convert_dict_code_for_predicate_if_necessary_impl(predicate);
+    }
+
+    for (auto predicate : _pre_eval_block_predicate) {
+        _convert_dict_code_for_predicate_if_necessary_impl(predicate);
+    }
+
+    std::set<const ColumnPredicate*> predicate_set {};
+    _opts.delete_condition_predicates->get_all_column_predicate(predicate_set);
```
Comment (wangbo) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:1212]: How about keep predicate_set as field of SegmentIterator so we don't need get them for every batch?

Comment (luozenglin) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp]: That sounds like a good idea. I have modified the code as you suggested. thank you.

**Comment by luozenglin** [comment]
> Please add regression test from the issue.

done

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (luozenglin) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp]: Yes, dev-1.1.2 has the same problem.

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**